StageEditor/Parser/DictionaryFile.cs

```diff
@@ -11,38 +11,66 @@ namespace Haven.Parser
 {
     public static class DictionaryFile
     {
-        public static readonly Dictionary<uint, string> Lookup = new Dictionary<uint, string>();
+        public static readonly Dictionary<uint, string> Alias = new();
+        public static readonly Dictionary<uint, string> Lookup = new();
+        public static int dictionaryUsed;
 
-        public static bool Load(string filename)
+        public static bool Load(string dictionaryFilename, string aliasFilename = "bin/dictionary-aliases.txt")
         {
             try
             {
-                var lines = File.ReadAllLines(filename);
+                if (File.Exists(dictionaryFilename))
+                {
+                    var lines = File.ReadAllLines(dictionaryFilename);
+
+                    foreach (var line in lines)
+                    {
+                        var cleanedLine = line.Trim();
+                        if (string.IsNullOrWhiteSpace(cleanedLine)) continue;
+
+                        var hash = Utils.HashString(cleanedLine);
+                        Lookup[hash] = cleanedLine;
+                    }
+                }
 
-                foreach (var line in lines)
+                if (File.Exists(aliasFilename))
                 {
-                    var hash = Utils.HashString(line.Replace("\n", ""));
-                    Lookup[hash] = line;
+                    var aliasLines = File.ReadAllLines(aliasFilename);
+                    foreach (var line in aliasLines)
+                    {
+                        var parts = line.Split(':');
+                        if (parts.Length != 2) continue;
+
+                        if (uint.TryParse(parts[0], System.Globalization.NumberStyles.HexNumber, null, out uint hash))
+                        {
+                            Alias[hash] = parts[1].Trim();
+                        }
+                    }
                 }
 
                 return true;
             }
-            catch(Exception e)
+            catch (Exception e)
             {
-                Log.Error("Failed to parse dictionary file {filename} \"{s}\"", filename, e.Message);
+                Log.Error("Failed to parse dictionary or alias file: {Message}", e.Message);
                 return false;
             }
         }
 
         public static string GetHashString(uint hash)
         {
-            if (Lookup.ContainsKey(hash))
+            if (Lookup.TryGetValue(hash, out var dictionaryValue))
+            {
+                dictionaryUsed = 1;
```

Comment:
We shouldn't be storing the dictionary type of the last fetched string this way. It's a bit messy.